modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java

```diff
@@ -425,17 +423,17 @@ private CompletableFuture<AsyncSqlCursor<List<Object>>> querySingle0(
 
                                 boolean implicitTxRequired = outerTx == null && rwOp;
 
-                                InternalTransaction implicitTx = implicitTxRequired ? txManager.begin() : null;
+                                InternalTransaction tx = implicitTxRequired ? txManager.begin()
+                                        : outerTx != null ? outerTx : new TransferredTxAttributesHolder(UUID.randomUUID(), clock.now());
```

Comment:
why you use UUID.randomUUID() here and not tx.id() ? is it correct ?

Reply:
oh, seems it correct for now, but it will fail distributed operations as i see.

Reply:
No, it will not fail distributed operations, because for read-only tx only timestamp matters.
see https://github.com/apache/ignite-3/blob/main/modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java#L884

The reason why I added random UUID here - because it seems to me better then passing `null` (since the real transaction must always have ID).
But may be you right, since in future this class will not implement transaction it can not have ID at all.